velox/vector/BaseVector.cpp

```diff
@@ -694,6 +696,17 @@ std::shared_ptr<BaseVector> BaseVector::createNullConstant(
         pool, size, true, type, UnscaledLongDecimal());
   }
 
+  if (type == HYPERLOGLOG() || type == JSON()) {
```

Comment:
We cannot list all custom types here. Let's discuss offline what would the right fix.

Reply:
We can simply check if it is a custom type I suppose
```
if (getType(type->toString, {})) {
}
```
The underlying problem is similar to Decimal Types. The TypeKind instead of Type is passed and we lose information.